### ISSUE TITLE
2021-10-31

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#641]
221至此我们通过对比学习，巧妙地在没有任何标签的情况下训练好了 SimCLR 模型，使得其Encoder的输出可以像正常有监督训练的模型一样表示图片的Representation信息。所以接下来就是利用这些 Representation的时候了，也就是在下游任务上Fine-tune。一旦 SimCLR 模型在对比学习任务上得到训练，它就可以用于迁移学习，如 ImageNet 分类，如下图19所示。此时在下游任务上 Fine-tune 模型时需要labeled data，但是数据量可以很小了。
222Self-Supervised Learning 超详细解读 (四)：MoCo系列解读 (1) https://zhuanlan.zhihu.com/p/382763210 (2021-10-20)
223总结下 Self-Supervised Learning 的方法，用 4 个英文单词概括一下就是：Unsupervised Pre-train, Supervised Fine-tune.
224第1个阶段不涉及任何下游任务，就是拿着一堆无标签的数据去预训练，没有特定的任务，这个话用官方语言表达叫做：in a task-agnostic way。第2个阶段涉及下游任务，就是拿着一堆带标签的数据去在下游任务上 Fine-tune，这个话用官方语言表达叫做：in a task-specific way; (2021-10-20)
2251.1 自监督学习的 Pretext Task; https://zhuanlan.zhihu.com/p/382763210 
226(1) BERT 的 Pretext Task：在训练 BERT 的时候，我们曾经在预训练时让它作填空的任务; 训练量很大; (2021-10-20)
227常见的损失函数是 cross entropy loss，它适合于数据的 label 是 one-hot 向量的形式。此时网络结构的最后一层是softmax，输出得到各个类的预测值。比如现在有3个类：dog, cat, horse，它们的 label 分别对应着 (1,0,0), (0,1,0), (0,0,1)，cross entropy loss会让 dog 图片的输出尽量接近 (1,0,0)，让 cat 图片的输出尽量接近 (0,1,0)，让 horse 图片的输出尽量接近 (0,0,1)。但是这也存在一个问题，就是假设再来3个类，分别是sky，car和bus。那么按道理dog与horse的距离 应该比dog与sky的距离近，因为dog与horse都属于动物；car与bus的距离 应该也比car与cat的距离近，因为car与bus都属于车类。但是 cross entropy loss确是一视同仁地把 dog与horse的距离 和 dog与sky的距离 看作是一样的。// Contrastive loss 的初衷是想让：1 相近的样本之间的距离越小越好。2 相远的样本之间的距离越大越好。
228如果神经网络的损失函数只满足条件1，那么网络会让任何的输入都输出相同的值，不论输入是 dog, cat, horse 还是 sky，car, bus，输出都是一样的，这确实满足了相近的样本之间的距离越小越好，但是却使得网络丧失了分类能力。// 如果神经网络的损失函数只满足条件 1 和 2，是不是就完善了呢？ // 实际上如果想让相远的样本之间的距离越大越好，就需要一个边界，否则如果 dog 是 (1,0,0)，那么 假设第1轮训练网络输出 cat 是 (0,1,0)，第2轮训练网络输出 cat 是 (0,5,0)...这样下去dog 与 cat 之间的距离越来越大，网络却没法收敛。
229Contrastive loss 改进的思路就是让相远的样本之间的距离越大越好，但是这个距离要有边界，即要求：1 相近的样本之间的距离越小越好。2 相远的样本之间的距离越大越好，这个距离最大是 [公式] 。如下图4弹簧图所示：黑色实心球代表与蓝色球相近的样本，白色空心球代表与蓝色球相远的样本，蓝色箭头的长度代表力的大小，方向代表力的方向。(a)：Contrastive loss 使得相近的样本接近。(b)：横轴代表样本之间的距离，纵轴代表loss值。Contrastive loss 使得相近的样本距离越小，loss值越低。(c)：Contrastive loss 使得相远的样本疏远。(d)：横轴代表样本之间的距离，纵轴代表loss值。Contrastive loss 使得相远的样本距离越大，loss值越低，但距离存在上界，就是红线与x轴的交点 [公式] ，代表距离的最大值。(e)：一个样本受到其他各个独立样本的作用，各个方向的力处于平衡状态，代表模型参数在 Contrastive loss 的作用下训练到收敛。
